### DISCUSSION TITLE
Simulator on linux: some annoying bugs

### DISCUSSION THREAD

**Discussion by melianmiko** [zepp-health/zeppos-forum#211] [Developer Feedback]
- Built-in version of QEMU depends on `libjpeg.so.8`, which is removed from Debian 12. Yes, this can be fixed with symlink, but anyway, this isn't normal...
- Simulator requires `sudo` access to run QEMU, but it didn't use PolicyKit, so password requests appear into terminal and is non-accessible.

![image](https://github.com/zepp-health/zeppos-forum/assets/67851274/cebaacd9-9c9c-4200-abc7-6f64d57e491f)

**Comment by zeppos-hx** [comment]
Yes, We have also identified these issues on Debian 12, we will fix them as soon as possible.
1. The new libjpeg62 cannot be symlinked to libjpeg.so.8 due to the version difference. To resolve this, it is necessary to compile the old version jpeg-8d and place it in /lib/x86_64-linux-gnu/, and then it will work properly
2. I think Policykit is the ideal solution, for now you can modify the /etc/sudoers to allow current user to use sudo without entering a password, like this

`%sudo	ALL=(ALL) NOPASSWD: ALL`

Here is the file of libjpeg8
[libjpeg.so.8.4.0.zip](https://github.com/zepp-health/zeppos-forum/files/12145442/libjpeg.so.8.4.0.zip)

**Reply by melianmiko** [reply]
Thank for information,
I've created a dockerfile to run simulator in separate environment as temporary solution, so it works.

But would be great if next versions will work out-of-box...
